core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -220,21 +251,53 @@ Iterable<Partition> all() {
 
   static class Partition {
     private final StructLike key;
-    private long recordCount;
-    private int fileCount;
     private int specId;
+    private long dataRecordCount;
+    private int dataFileCount;
+
+    private final Set<DeleteFile> equalityDeleteFiles;
+    private final Set<DeleteFile> positionDeleteFiles;
 
     Partition(StructLike key) {
       this.key = key;
-      this.recordCount = 0;
-      this.fileCount = 0;
       this.specId = 0;
+      this.dataRecordCount = 0;
+      this.dataFileCount = 0;
+      this.positionDeleteFiles = Sets.newHashSet();
+      this.equalityDeleteFiles = Sets.newHashSet();
+    }
+
+    private void update(FileScanTask task) {
+      this.dataRecordCount += task.file().recordCount();
+      this.dataFileCount += 1;
+      this.specId = task.file().specId();
+      if (TableScanUtil.hasDeletes(task)) {
```

Comment:
Why not just, if task.deletes().size() > 0?   I think we are getting FileScanTask here, not CombinedScanTask, isnt it?

Reply:
I think it internally calls exactly that. So, no need to change? 
https://github.com/apache/iceberg/blob/fa0bcdf2593bd70425a4cae0148b26f96b1e46b1/core/src/main/java/org/apache/iceberg/util/TableScanUtil.java#L68-L70

Reply:
You are right, I was reading the other TableScanUtil.hasDeletes method